src/mrinufft/trajectories/trajectory3D.py

```diff
@@ -149,6 +150,180 @@ def initialize_3D_cones(
     return trajectory.reshape((Nc, Ns, 3))
 
 
+def initialize_3D_helical_shells(Nc, Ns, nb_shells, oversampling=1, tilt="uniform"):
+    """Initialize 3D trajectories with helical shells.
+
+    Parameters
+    ----------
+    Nc : int
+        Number of shots
+    Ns : int
+        Number of samples per shot
+    nb_shells : int
+        Number of concentric shells/spheres
+    oversampling : int
+        Multiplicative factor for Ns without changing the curves, by default 1
+    tilt : str, float, optional
+        Angle between shots over a shell surface, by default "uniform"
+
+    Returns
+    -------
+    array_like
+        3D helical shell trajectory
+    """
+    # Check arguments validity
+    if Nc < nb_shells:
+        raise ValueError("Argument nb_shells should not be higher than Nc.")
+    trajectory = np.zeros((Nc, Ns * oversampling, 3))
+
+    # Attribute shots to shells following a prescribed density
+    Nc_per_shell = np.ones(nb_shells).astype(int)
+    density = np.array(
+        [
+            int(np.ceil(8 * np.pi / np.sqrt(3) * (i + 1) ** 2 / Ns))
+            for i in range(nb_shells)
+        ]
+    )
+    for _ in range(Nc - nb_shells):
+        idx = np.argmax(density / Nc_per_shell)
+        Nc_per_shell[idx] += 1
+
+    # Create shells one by one
+    count = 0
+    radii = (0.5 + np.arange(nb_shells)) / nb_shells
+    for i in range(nb_shells):
+        # Prepare shell parameters
+        Ms = Nc_per_shell[i]
+        k0 = radii[i]
+        kz = k0 * np.linspace(-1, 1, Ns * oversampling)
+
+        # Initialize first shot cylindrical coordinates
+        magnitudes = k0 * np.sqrt(1 - (kz / k0) ** 2)
+        angles = (
+            np.sqrt(Ns * np.pi / Ms) * np.arcsin(kz / k0) + 2 * np.pi * (i + 1) / Ms
+        )
+
+        # Format first shot into trajectory
+        trajectory[count, :, 0] = magnitudes * np.cos(angles)
+        trajectory[count, :, 1] = magnitudes * np.sin(angles)
+        trajectory[count : count + Ms, :, 2] = kz[None]
+
+        # Rotate shots to create the shell
+        rotation = R2D(initialize_tilt(tilt, Ms))
+        for j in range(1, Ms):
+            trajectory[count + j, :, :2] = trajectory[count + j - 1, :, :2] @ rotation
+        count += Ms
+    return KMAX * trajectory
+
+
+def initialize_3D_annular_shells(
+    Nc, Ns, nb_shells, shell_tilt=np.pi, ring_tilt=np.pi / 2
+):
+    """Initialize 3D trajectories with annular shells.
+
+    Parameters
+    ----------
+    Nc : int
+        Number of shots
+    Ns : int
+        Number of samples per shot
+    nb_shells : int
+        Number of concentric shells/spheres
+    shell_tilt : str, float, optional
+        Angle between consecutive shells, by default pi
+    ring_tilt :
+        Angle controlling approximately the shell halves rotation, by default pi / 2
+
+    Returns
+    -------
+    array_like
+        3D annular shell trajectory
+    """
```

Comment:
Don't forget to add reference here also.

Reply:
Done